genrl/deep/agents/sac/sac.py

```diff
@@ -121,27 +129,36 @@ def create_model(self) -> None:
         Initialize the model
         Initializes optimizer and replay buffers as well.
         """
-        state_dim, action_dim, discrete, _ = get_env_properties(self.env)
+        if isinstance(self.network, str):
+            state_dim, action_dim, discrete, _ = get_env_properties(self.env)
 
-        self.q1 = (
-            get_model("v", self.network_type)(state_dim, action_dim, "Qsa", self.layers)
-            .to(self.device)
-            .float()
-        )
+            self.q1 = (
+                get_model("v", self.network)(state_dim, action_dim, "Qsa", self.layers)
+                .to(self.device)
+                .float()
+            )
 
-        self.q2 = (
-            get_model("v", self.network_type)(state_dim, action_dim, "Qsa", self.layers)
-            .to(self.device)
-            .float()
-        )
+            self.q2 = (
+                get_model("v", self.network)(state_dim, action_dim, "Qsa", self.layers)
+                .to(self.device)
+                .float()
+            )
 
-        self.policy = (
-            get_model("p", self.network_type)(
-                state_dim, action_dim, self.layers, discrete, False, sac=True
+            self.policy = (
+                get_model("p", self.network)(
+                    state_dim, action_dim, self.layers, discrete, False, sac=True
+                )
+                .to(self.device)
+                .float()
             )
-            .to(self.device)
-            .float()
-        )
+        else:
+            self.model = self.network
+            assert "value" and "policy" in dir(
+                self.model
+            ), "network must contain value and policy attributes"
+            self.q1 = self.model.value.to(self.device).float()
+            self.q2 = self.model.value.to(self.device).float()

```

Comment:
This is incorrect. The two value models are supposed to be different. Need not worry because SAC needs to be refactored but for the time being maybe change `value` to `q1` and `q2`